lntest/itest/lnd_macaroons_test.go

```diff
@@ -371,6 +371,29 @@ func testBakeMacaroon(net *lntest.NetworkHarness, t *harnessTest) {
 			require.Error(t, err)
 			require.Contains(t, err.Error(), "permission denied")
 		},
+	}, {
+		// Seventh test: check that if the allow_external_permissions
+		// flag is set, we are able to feed BakeMacaroons permissions
+		// that LND is not familiar with.
+		name: "allow external macaroon permissions",
+		run: func(ctxt context.Context, t *testing.T,
+			adminClient lnrpc.LightningClient) {
+
+			// We'll try a permission from Pool to test that the
+			// allow_external_permissions flag properly allows it.
+			rootKeyID := uint64(4200)
+			req := &lnrpc.BakeMacaroonRequest{
+				RootKeyId: rootKeyID,
+				Permissions: []*lnrpc.MacaroonPermission{{
+					Entity: "account",
+					Action: "read",
+				}},
+				AllowExternalPermissions: true,
+			}
+
+			_, err := adminClient.BakeMacaroon(ctxt, req)
```

Comment:
Should we also look at the macaroon and check that the permission was actually added?

Reply:
@guggero sounds good, how do we look at the macaroon's permissions?

Reply:
You can decode the macaroon ID and look at the permissions contained within: https://github.com/lightningnetwork/lnd/blob/master/cmd/lncli/cmd_macaroon.go#L387

Reply:
brilliant, thanks!